core/src/main/java/org/jdbi/v3/core/statement/SqlStatement.java

```diff
@@ -111,32 +117,28 @@ public This setQueryTimeout(final int seconds)
     }
 
     /**
-     * Close the handle when the statement is closed.
-     *
-     * @return the same Query instance
+     * When the statement is closed, commit it and then close the owning Handle.
```

Comment:
This isn't exactly accurate since the commit happens on the handle, not the statement (so it can encompass more than this statement). How about something like

*Transfer ownership of the handle to the statement: when the statement is closed, commit the handle's transaction (if one exists) and close the handle.*